bgls/near_clifford_solver.py

```diff
@@ -0,0 +1,93 @@
+# Licensed under the Apache License, Version 2.0 (the "License");
```

Comment:
There needs to be a `near_clifford_solver_test.py` file testing these functions.

Reply:
added some tests, made a simple circuit with one T and assert the output is two circuits with it replaced by I and S respectively. I can expand to more complicated/thorough tests as well

Reply:
Looks good for now, up to a few new comments!